### PR TITLE
Close GUI before daemon is killed when upgrading app version

### PR DESCRIPTION
**Pull request by MarkusPettersson98**
Closing the GUI shows the disconnected and unsecure notification and the tray icon switches to the unlocked state when it loses the connection to the daemon. For versions where the GUI don't disconnect when killed (`2022.5` and later) we want to kill the GUI first, but we want to keep it like it currently is when upgrading from older versions.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)

### CODE REVIEW

**Review by faern — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: all files reviewed, 3 unresolved discussions (waiting on @MarkusPettersson98)

___
*[`dist-assets/pkg-scripts/preinstall` line 44 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHiDsg4W_F0RD_3zH4:-NYHiDshAZrq3lU-90Dc:b-mkygfj) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/preinstall#L44)):*
> ```
> KILL_GUI_BEFORE_DAEMON=$?
> if [[ $KILL_GUI_BEFORE_DAEMON -eq 0 ]]
> then
> ```

Please follow coding guidelines. Both with `then`-placement and indentation. https://github.com/mullvad/coding-guidelines/blob/main/bash.md

**Review by MarkusPettersson98 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: 0 of 2 files reviewed, 4 unresolved discussions (waiting on @faern and @raksooo)

___
*[`dist-assets/pkg-scripts/postinstall` line 80 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHhbrK3HRd1FTxnVVs:-NYHln4TBzRzOmyLufDG:b-fip99d) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/postinstall#L80)):*
<details><summary><i>Previously, raksooo (Oskar Nyberg) wrote…</i></summary><blockquote>

Maybe we should remove/change this error message since it's expected to happen when upgrading from anything >2022.5?
</blockquote></details>

I see how this could be considered confusing. Do we remove it entirely, or do we augment it by saying that it is expected when upgrading from anything >2022.5? :wink:

**Review by faern — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @MarkusPettersson98 and @raksooo)

___
*[`dist-assets/pkg-scripts/postinstall` line 80 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHhbrK3HRd1FTxnVVs:-NYI21Ou7dfa7Yb4jhx3:bv4iz9a) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/postinstall#L80)):*
<details><summary><i>Previously, MarkusPettersson98 (Markus Pettersson) wrote…</i></summary><blockquote>

I see how this could be considered confusing. Do we remove it entirely, or do we augment it by saying that it is expected when upgrading from anything >2022.5? :wink: 
</blockquote></details>

To be nitpicky, because otherwise this becomes hard to parse later when we actually want to remove the code: I think both of mean `>=2022.5`.

I think it can be removed entirely. It's not like it's very important. I don't think we have debugged any issue with the help of this output actually. And the vast majority of people will upgrade from `>=2022.5` anyway.
___
*[`dist-assets/pkg-scripts/preinstall` line 40 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHidDv7rD263lKcfyT:-NYI1hI-1NCCYvP2SE-c:b6zeqf0) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/preinstall#L40)):*
Maybe add to this script about how this could be simplified some time after dropping support for `2022.5` and just always kill the GUI here instead of having the check.

I think the comment in postinstall is still a bit strange. It says the GUI *can* be killed in preinstall, but it already is being killed in preinstall thanks to this PR. And it's a bit backwards to say "Kill daemon after GUI" attached to a piece of code that kills the GUI. Can I suggest something along the lines of:

> For compatibility with `<2022.5`, kill the GUI after the upgrade. We now do this in preinstall instead. This code can be removed when we drop support for `<2022.5`.

**Review by MarkusPettersson98 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @faern and @raksooo)

___
*[`dist-assets/pkg-scripts/postinstall` line 80 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHhbrK3HRd1FTxnVVs:-NYITCW01a8aEag6Yugh:brxwg7x) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/postinstall#L80)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

To be nitpicky, because otherwise this becomes hard to parse later when we actually want to remove the code: I think both of mean `>=2022.5`.

I think it can be removed entirely. It's not like it's very important. I don't think we have debugged any issue with the help of this output actually. And the vast majority of people will upgrade from `>=2022.5` anyway.
</blockquote></details>

I removed the error message from `postinstall`, but not `preinstall`. Just tell me if you think it should be removed from `postinstall` as well , but at least with this change we shouldn't see any false positives when upgrading an app with version `>=2022.5` :blush: 
___
*[`dist-assets/pkg-scripts/preinstall` line 40 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHidDv7rD263lKcfyT:-NYITBJzCHjERylB-x-3:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/preinstall#L40)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Maybe add to this script about how this could be simplified some time after dropping support for `2022.5` and just always kill the GUI here instead of having the check.

I think the comment in postinstall is still a bit strange. It says the GUI *can* be killed in preinstall, but it already is being killed in preinstall thanks to this PR. And it's a bit backwards to say "Kill daemon after GUI" attached to a piece of code that kills the GUI. Can I suggest something along the lines of:

> For compatibility with `<2022.5`, kill the GUI after the upgrade. We now do this in preinstall instead. This code can be removed when we drop support for `<2022.5`.
</blockquote></details>

Done.

**Review by faern — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: all files reviewed, 2 unresolved discussions (waiting on @raksooo)

___
*[`dist-assets/pkg-scripts/postinstall` line 80 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHhbrK3HRd1FTxnVVs:-NYIXxA49LU29hA2K7iP:bcoyx84) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/postinstall#L80)):*
<details><summary><i>Previously, MarkusPettersson98 (Markus Pettersson) wrote…</i></summary><blockquote>

I removed the error message from `postinstall`, but not `preinstall`. Just tell me if you think it should be removed from `postinstall` as well , but at least with this change we shouldn't see any false positives when upgrading app with version `>=2022.5` :blush: 
</blockquote></details>

I think you still need `|| continue`. When you removed the `||` any error will immediately terminate the entire script (`set -e`) when any command in it exits with an error.

**Review by MarkusPettersson98 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @faern and @raksooo)

___
*[`dist-assets/pkg-scripts/postinstall` line 80 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHhbrK3HRd1FTxnVVs:-NYIYyQACxkWLMTosyXg:bhpl7ht) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/postinstall#L80)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

I think you still need `|| continue`. When you removed the `||` any error will immediately terminate the entire script (`set -e`) when any command in it exits with an error.
</blockquote></details>

`|| continue` is only valid in loops, but `|| true` should do the trick, no?

**Review by faern — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: all files reviewed, 2 unresolved discussions (waiting on @raksooo)

___
*[`dist-assets/pkg-scripts/postinstall` line 80 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799#-NYHhbrK3HRd1FTxnVVs:-NYIZy0eAjBXbE3rq-O_:bp0j8l) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b9f5e1557ceaf779544bdfb0521a4b88c23a77e9/dist-assets/pkg-scripts/postinstall#L80)):*
<details><summary><i>Previously, MarkusPettersson98 (Markus Pettersson) wrote…</i></summary><blockquote>

`|| continue` is only valid in loops, but `|| true` should do the trick, no?
</blockquote></details>

Yeah, sorry. That's what I meant. Great :+1:

**Review by faern — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by raksooo — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4799)* status: :shipit: complete! all files reviewed, all discussions resolved
